Specs/Core/sampleTerrainMostDetailedSpec.js

```diff
@@ -74,7 +74,7 @@ defineSuite([
     });
 
     it('works for a dodgy point right near the edge of a tile', function() {
-        var positions = [new Cartographic(0.33179290856829535, 0.7363107781851078)];
+        var positions = [new Cartographic(0.33179290856829535, 0.7363107781851076)];
```

Comment:
The slight change in this spec is kind of scary. Does the spec fail without the change? Do we know why the previous implementation needed this spec to pass as-is?

Reply:
Previous implementation was added here: https://github.com/AnalyticalGraphicsInc/cesium/commit/22ff240f5c6d60742282dd175897e50332f087b7.

Reply:
Does this spec fail if the value is reverted? The linked commit seems to suggest that it was important for this spec to pass with the given value.

Reply:
It looks like the `clamp` behavior is preserved in the current changes too, which may suggest that there are some positions that were inside the triangle as-per the barycentric method but not according to the raycasting method.

Reply:
I agree with @likangning93, changing this spec here is the wrong thing to do.  It was put here specifically because we were seeing corner cases fail.

Reply:
Here's a thought: can we make the raycast check more permissive than the barycentric method but still use the barycentric method for points that pass the raycast check?

Reply:
I just ran a small test and it looks like the triangle isn't checking for points on the triangle. I'm going to look into how we can include this.

Reply:
@likangning93 I reverted the spec change and checked for points on the triangle. With that change the benchmark changed from 1920ms to 2500ms (ouch).

Reply:
> Here's a thought: can we make the raycast check more permissive than the barycentric method but still use the barycentric method for points that pass the raycast check?

I don't think it would hurt, because technically it will only be done for one triangle. But then again, it is redundant. The check is stringent enough to not allow points outside the triangle from getting int.

Reply:
I don't think it's that redundant, it's similar to using a BVH.
The problem is that the raycast-based check has different tolerances than the barycentric check, but the barycentric check has been calibrated for corner cases that we'd need to re-evaluate.
Making the raycast check slightly more permissive would eliminate the urgency around those corner cases because they'd still be handled by the barycentric check.